yapapi/contrib/strategy/rep_a1.py

```diff
@@ -0,0 +1,107 @@
+from typing import Dict, Set, TYPE_CHECKING
+from decimal import Decimal
+from collections import defaultdict
+import logging
+from random import gauss
+
+from yapapi.strategy import WrappingMarketStrategy
+from yapapi import events
+
+if TYPE_CHECKING:
+    from yapapi.rest.payment import DebitNote, Invoice
+    from yapapi.rest.market import OfferProposal
+
+
+logger = logging.getLogger(__name__)
+
+ActivityId = str
+AgreementId = str
+
+
+async def get_provider_score(provider_id: str) -> float:
+    #   This is a mock.
+    #   In the final (alpha1) version this will query the API for the provider score,
+    #   but the score received will have the same "meaning", i.e.: how this provider
+    #   differs from the average provider quality, expressed in the number of SDs.
+    return gauss(0, 1)
+
+
+class RepA1(WrappingMarketStrategy):
+    def __init__(self, base_strategy):
+        super().__init__(base_strategy)
+
+        self._failed_activities: Set[ActivityId] = set()
+        self._accepted_amounts: Dict[ActivityId, Decimal] = defaultdict(Decimal)
+        self._agreement_activity_map: Dict[AgreementId, Set[ActivityId]] = defaultdict(set)
+
+    #################
+    #   OFFER SCORING
+    async def score_offer(self, offer: "OfferProposal") -> float:
+        offer_score = await super().score_offer(offer)
+        provider_score = await get_provider_score(offer.issuer)
+        combined_score = self._final_score(offer_score, provider_score)
+
+        provider_name = offer._proposal.proposal.properties['golem.node.id.name']
```

Comment:
nitpicking.... maybe it would make sense for this property key to be added as a constant somewhere (`yapapi/props/__init__.py` ?) ... ?

Reply:
I'll check `provider_view` etc, if I don't find this I'll leave it as it is

Reply:
I couldn't find this easily. Let's leave this for the future (either way, it's just for logging).